From da0990bf05e3433aa21c50e6a62f33fe6649b417 Mon Sep 17 00:00:00 2001
From: Niklas Haas <git@haasn.xyz>
Date: Fri, 14 Dec 2018 14:16:10 +0100
Subject: [PATCH] Work-around meson prefix bug

Normally this is inferred from the environment's CFLAGS/LDFLAGS, but old
meson versions don't do this correctly. So work-around it by explicitly
including the prefix when searching for glslang.
---
 src/meson.build | 23 +++++++++++++++--------
 1 file changed, 15 insertions(+), 8 deletions(-)

diff --git a/src/meson.build b/src/meson.build
index 411ce2c..bc9214a 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -48,6 +48,11 @@ glslang_combined = disabler()
 glslang_min_ver = 2763
 glslang_req = get_option('glslang')
 
+# fix build inside contrib dir (cf. meson issue #4597)
+prefix_option = get_option('prefix')
+prefix_lib = prefix_option + '/lib'
+prefix_include = include_directories(prefix_option + '/include')
+
 if glslang_req.auto() and shaderc.found()
 
   # we only need one or the other, and shaderc is preferred
@@ -58,12 +63,12 @@ if glslang_req.auto() and shaderc.found()
 else
 
   glslang_deps = [
-    cxx.find_library('glslang',     required: glslang_req),
-    cxx.find_library('HLSL',        required: glslang_req),
-    cxx.find_library('OGLCompiler', required: glslang_req),
-    cxx.find_library('OSDependent', required: glslang_req),
-    cxx.find_library('SPIRV',       required: glslang_req),
-    cxx.find_library('SPVRemapper', required: glslang_req),
+    cxx.find_library('glslang',     dirs: prefix_lib, required: glslang_req),
+    cxx.find_library('HLSL',        dirs: prefix_lib, required: glslang_req),
+    cxx.find_library('OGLCompiler', dirs: prefix_lib, required: glslang_req),
+    cxx.find_library('OSDependent', dirs: prefix_lib, required: glslang_req),
+    cxx.find_library('SPIRV',       dirs: prefix_lib, required: glslang_req),
+    cxx.find_library('SPVRemapper', dirs: prefix_lib, required: glslang_req),
   ]
 
   glslang_found = true
@@ -75,7 +80,8 @@ endif
 
 if glslang_found
   glslang_ver = cxx.get_define('GLSLANG_PATCH_LEVEL',
-      prefix: '#include <glslang/Include/revision.h>'
+      prefix: '#include <glslang/Include/revision.h>',
+      include_directories : prefix_include
   ).to_int()
 
   if glslang_ver >= glslang_min_ver
@@ -85,7 +91,8 @@ if glslang_found
     # -pthread) doesn't work. We actually need -lpthreads for the glslang
     # object files to link, for whatever weird reason.
     pthread = cxx.find_library('pthread', required: false)
-    glslang_combined = declare_dependency(dependencies: glslang_deps + [pthread])
+    glslang_combined = declare_dependency(dependencies: glslang_deps + [pthread],
+        include_directories : prefix_include)
   else
     error('glslang revision @0@ too old! Must be at least @1@'
           .format(glslang_ver, glslang_min_ver))
-- 
2.19.2

